File: Lesson1/src/main/java/com/mikeabney/katas/dice/domain/Die.java
package com.mikeabney.katas.dice.domain;

import java.util.Random;

public class Die {
    private Random generator;
    private int sides;

    public Die(int sides) {
        if (sides < 1) {
            throw new IllegalArgumentException(String.format("Attempt to create a die with negative sides [%i].", sides));
        }
        this.sides = sides;
        generator = new Random(System.currentTimeMillis());
    }

    public int roll() {
        return generator.nextInt(sides) + 1;
    }
}

File: Lesson1/Dice/Dice/Die.cs
using System;

namespace Dice
{
    internal class Die : IDie
    {
        private readonly int _sides;

        public Die(int sides)
        {
            _sides = sides;
        }

        public int Roll()
        {
            var rnd = new Random();
            return rnd.Next(1, this._sides + 1);
        }
    }
}